cmd/faFindFast/faFindFast.go

```diff
@@ -6,29 +6,69 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
 	"log"
 	"math"
+	"os"
+	"runtime/pprof"
 
-	"github.com/vertgenlab/gonomics/exception"
 	"github.com/vertgenlab/gonomics/fasta"
-	"github.com/vertgenlab/gonomics/fileio"
 )
 
-func faFindFast(inFile string, outFile string, windowSize int, chromName string, removeN bool, longOutput bool, divergenceRate float64) {
-	records := fasta.Read(inFile)
+type Settings struct {
+	InFile          string
+	OutFile         string
+	FirstQueryName  string
+	SecondQueryName string
+	WindowSize      int
+	RefChromName    string
+	RemoveN         bool
+	LongOutput      bool
+	DivergenceRate  float64
+	OutputAlnPos    bool
+}
+
+func faFindFast(s Settings) {
+	var reference, firstQuery, secondQuery []dna.Base
+	var found bool
```

Comment:
- You can also check against the map if the key exists key, which will return a bool without using the found flag.